tests/lib/rules/no-unneeded-ternary.js

```diff
@@ -26,6 +26,7 @@ ruleTester.run("no-unneeded-ternary", rule, {
         "var a = x === 2 ? 'Yes' : false;",
         "var a = x === 2 ? 'true' : 'false';",
         "var a = foo ? foo : bar;",
+        "var value = 'a';var canSet = true;var result = value || (canSet ? 'unset' : 'can not set')",
```

Comment:
Sorry, I did a lot of fat fingering in that comment.

Your fix logic potentially parenthesizes both the node's `test` and `alternate`. I only see a new test for the `alternate`. Is there a test that covers parenthesizing of the `test`?